### ISSUE TITLE
Window content is too small to be readable

### ISSUE THREAD

**Issue by TheCasualObserver** [shpaass/yafc-ce#255]
- Tested with YAFC-CE 8.1, 8.0, 7.5 and 7.0
- Debian Linux 12
- Xfce 4.18
- single display attached (1920x1080)
```
~ dotnet --list-runtimes
Microsoft.AspNetCore.App 8.0.8 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 8.0.8 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

When I start YAFC, the window is very small and the text unreadable as shown in the following screenshot

![image](https://github.com/user-attachments/assets/c28ec3b0-d5e4-4354-b249-f217e3446b0b)

The window is not resizable, so I can't make it bigger.
I would expect, that the text in the window is readable.

**Comment by shpaass** [comment]
Is the main screen the same smol size as the welcome screen?

To reach the main screen, you can 
1. type any project name in the first line, 
2. press the blue part on the second line and select the Factorio data folder,
3. leave the third line empty to make a vanilla project. 
4. press Enter to start the loading.

**Comment by TheCasualObserver** [comment]
Thanks for the instructions.

Clicking on the blue button in the second line opens a new window:
![image](https://github.com/user-attachments/assets/3051b106-20b4-4b28-bf5f-aeae519ac814)

After pasting the path of the data folder directory into the field it looks like I can open the main screen:
![image](https://github.com/user-attachments/assets/2c686440-b0c6-43eb-bc3b-f306ff82ab56)

I can resize this Window, but the content still stays small:
![image](https://github.com/user-attachments/assets/1f4ffb61-433e-4f77-8abc-2bae24655f0d)

**Comment by DaleStan** [comment]
This sounds similar to #239. What DPI does your monitor report?

**Comment by TheCasualObserver** [comment]
I have seen that issue, but didn't consider it relevant, because it is related to hidpi.

My display is a projector and `xrandr` reports
```
~ xrandr
Screen 0: minimum 8 x 8, current 1920 x 1080, maximum 32767 x 32767
HDMI-0 disconnected (normal left inverted right x axis y axis)
DP-0 disconnected (normal left inverted right x axis y axis)
DP-1 disconnected (normal left inverted right x axis y axis)
DP-2 disconnected (normal left inverted right x axis y axis)
DP-3 disconnected (normal left inverted right x axis y axis)
HDMI-1 connected 1920x1080+0+0 (normal left inverted right x axis y axis) 2210mm x 1250mm
   1920x1080     60.00*+  59.94    50.00    23.98  
   1400x1050     59.98  
   1280x1024     60.02  
   1280x960      60.00  
   1280x768      59.87  
   1280x720      59.94    50.00  
   1024x768      60.00  
   800x600       60.32  
   720x576       50.00  
   720x480       59.94  
   640x480       59.94    59.93
```

After detaching the projector and attaching a monitor, the content of the window is displayed correctly.
So I guess this is related to the reported dimensions of `2210mm x 1250mm`.

Is there a way for me to configure the dpi manually?

**Comment by TheCasualObserver** [comment]
I was able to fix this bug for me with the following change. I believe it's possible, that a change like this would also help with #239.
Based on https://www.notebookcheck.net/DPI-Fineness-of-Displays.1310.0.html it looks like 96 might be a reasonable number for distinguishing between actual dpi and incorrectly reported values.

``` diff
diff --git a/Yafc.UI/Core/Window.cs b/Yafc.UI/Core/Window.cs
index eb2f0e2..7d217e8 100644
--- a/Yafc.UI/Core/Window.cs
+++ b/Yafc.UI/Core/Window.cs
@@ -70,7 +70,7 @@ public abstract class Window : IDisposable {
         _ = SDL.SDL_GetDisplayDPI(display, out float dpi, out _, out _);
         _ = SDL.SDL_GetDisplayBounds(display, out var rect);
         // 82x60 is the minimum screen size in units, plus some for borders
-        int desiredUnitsToPixels = dpi == 0 ? 13 : MathUtils.Round(dpi / 6.8f);
+        int desiredUnitsToPixels = dpi < 96 ? 13 : MathUtils.Round(dpi / 6.8f);
 
         if (desiredUnitsToPixels * 82f >= rect.w) {
             desiredUnitsToPixels = MathUtils.Floor(rect.w / 82f);
```

**Comment by kkartaltepe** [comment]
>So I guess this is related to the reported dimensions of 2210mm x 1250mm.

Yes this data comes from the EDID of your monitor/projector. And is usually for the largest model of display that the manufacturer ships that firmware on (in my experience).

96 is the historical default display dpi in X11 and so the default 100% scaling. Similar anything less than this is likely a firmware bug/manufacture laziness, and its unlikely anyone actually wants the text to be smaller even on a low dpi display (especially given the poor rendering quality of the ui toolkit chosen).
